lib/scholar/neighbors/utils.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Scholar.Neighbors.Utils do
+  import Nx.Defn
+  require Nx
+
+  defn find_neighbors(query, data, candidate_indices, opts) do
```

Comment:
Quick question: does it work on any query/data/candidate_indices or it has to be related to random forests? We probably don't want to make this public for now (hence `@moduledoc false`), but we should most likely add a `@doc` entry to this one as well, explaining the shape and the role of each argument.

Reply:
Yeah, it is kinda general. For example, brute-force k-NN search can be implemented as `find_neighbors(query, tensor, tensor, num_neighbors: k)`. Two things that should be added:
- More distances other than Euclidean.
- A flag whether there is a need to remove duplicates from candidate lists or not. For example, brute-force search as specified above doesn't need duplicate removal. Similarly for random projection forest when there is a single tree (there can be no duplicates in a leaf). Or should there be two functions: one that removes the duplicates and the other that does not?

I called the function `find_neighbors` simply because I cannot think of a better name. Something like `find_nearest_neighbors` or `nn_search` might be better.